 logd: simpleperf inspired optimizations

memset is hot on log writes with too much overhead, not required.

diff --git a/logd/LogListener.cpp b/logd/LogListener.cpp
index 000d2b1..13fc5fa 100644
--- a/logd/LogListener.cpp
+++ b/logd/LogListener.cpp
@@ -44,7 +44,6 @@ bool LogListener::onDataAvailable(SocketClient *cli) {
     char buffer[sizeof_log_id_t + sizeof(uint16_t) + sizeof(log_time)
         + LOGGER_ENTRY_MAX_PAYLOAD];
     struct iovec iov = { buffer, sizeof(buffer) };
-    memset(buffer, 0, sizeof(buffer));
 
     char control[CMSG_SPACE(sizeof(struct ucred))];
     struct msghdr hdr = {
@@ -59,6 +58,9 @@ bool LogListener::onDataAvailable(SocketClient *cli) {
 
     int socket = cli->getSocket();
 
+    // To clear the entire buffer is secure/safe, but this contributes to 1.68%
+    // overhead under logging load. We are safe because we check counts.
+    // memset(buffer, 0, sizeof(buffer));
     ssize_t n = recvmsg(socket, &hdr, 0);
     if (n <= (ssize_t)(sizeof(android_log_header_t))) {
         return false;
